cuda-compatible-runtime.spec

```diff
@@ -1,18 +1,22 @@
 ### RPM external cuda-compatible-runtime 1.0
 
 %define branch master
-%define commit 18d5a51bfb32fbb7b765dd2eecd14e193cce8126
+%define commit bfe5537537428ab4a72ae929c77977a55501c576
 
-Source: git+https://:@gitlab.cern.ch:8443/cms-patatrack/%{n}.git?obj=%{branch}/%{commit}&export=%{n}&filter=./test.c&output=/%{n}-%{realversion}.tgz
+Source: git+https://:@gitlab.cern.ch:8443/cms-patatrack/%{n}.git?obj=%{branch}/%{commit}&export=%{n}&filter=./test.cu&output=/%{n}-%{realversion}.tgz
 Requires: cuda
 AutoReq: no
 
 %prep
 %setup -n %{n}
 
 %build
+## INCLUDE cuda-flags
+# defines nvcc_stdcxx and cuda_flags_4
+
 rm -rf %{_builddir}/build && mkdir %{_builddir}/build
-gcc -std=c99 -O2 -Wall test.c -I $CUDA_ROOT/include -L $CUDA_ROOT/lib64 -L $CUDA_ROOT/lib64/stubs -l cudart_static -l cuda -ldl -lrt -pthread -static-libgcc -o %{_builddir}/build/cuda-compatible-runtime # || true
+$CUDA_ROOT/bin/nvcc %{nvcc_stdcxx} -O2 -g %{cuda_flags_4} test.cu -I $CUDA_ROOT/include -L $CUDA_ROOT/lib64 -L $CUDA_ROOT/lib64/stubs --cudart static -ldl -lrt --compiler-options '-Wall -pthread' -o %{_builddir}/build/cuda-compatible-runtime # || true
```

Comment:
@fwyzard , isn't it going to fail if gcc version not compatible with cuda?

Reply:
@fwyzard ping

Reply:
> @fwyzard , isn't it going to fail if gcc version not compatible with cuda?

Yes, good point.
If that happens, there really is no version of the runtime we want to use...

Reply:
@fwyzard , should we fix this so that it does not fail the build process? I mean in that case we should still create cuda-compatible-runtime which exit with non-zero code e.g.

```
if [ $(%{cuda_gcc_support}) = true ] ; then
  $CUDA_ROOT/bin/nvcc %{nvcc_stdcxx} -O2 -g %{cuda_flags_4} test.cu -I $CUDA_ROOT/include -L $CUDA_ROOT/lib64 -L $CUDA_ROOT/lib64/stubs --cudart static -ldl -lrt --compiler-options '-Wall -pthread' -o %{_builddir}/build/cuda-compatible-runtime
else
  echo "CUDA ${CUDA_VERSION} is not compatiable with GCC ${GCC_VERSION}" > %{_builddir}/build/cuda-compatible-runtime
  echo "false" >> %{_builddir}/build/cuda-compatible-runtime
  chmod +x %{_builddir}/build/cuda-compatible-runtime
fi
```

Reply:
@fwyzard , what do you think about https://github.com/cms-sw/cmsdist/pull/7419/files#r741837428 ?

Reply:
sorry, I forgot :-(
let me see...

Reply:
If it works, sounds like a good idea.
Can we actually use `%{cuda_gcc_support}` before CUDA has been set up ?

I guess I can make the changes and try...

Reply:
In fact, the `%install` part was already setting up a symlink to `/usr/bin/false` is the build failed; but I like the idea of a more verbose message.

Reply:
as an alternative - is there a way to use the system compiler for building this, instead of the one bundled with CMSSW ?